src/server/routes/pathways/generate-pathway-json/biopax-metadata/index.js

```diff
@@ -35,18 +36,31 @@ let extractBiopaxMetadata = biopaxJsonEntry => {
     type: type.includes(':') ? type.split(':')[1] : type,
     standardName,
     displayName,
-    databaseIds
+    xrefLinks
   };
 };
 
+// helper for converting xrefs to map of uris grouped by db
+const xref2link = ( db, id, xrefLinks ) => {
```

Comment:
Generally I think that it is not a good idea to have functions like this produce side effects.  In my opinion it makes the code harder to understand.  
As a general guideline going forward I think it is good to keep functions that:
1. Are generally independent reusable pieces of functionality.
2. Have clear inputs and outputs.
3. Have Minimal side effects.
 
If I was reading this function for the first time I would be confused and ask:
1.  Why does this function update a `xrefLinks` object?
2.  Where is this function being used?
3.  What is the context of this function?  

The answer to my knowledge is that it is being used only in one place and the reason xrefLinks is being updated is because it is preprocessing all xrefs so that they can be easily accessed via each element id.  I think this information is obscured in the `xref2Link` function and therefore I think it is better if this code is just inlined into the`biopaxText2ElementMap` function.

i.e.

```
// process all xrefId elements and get a uri for each of them
xrefIds.fiilter( xrefId => xrefId != null ).forEach( xrefId => {
  let { k, v } = xRefMap.get( xrefId );
  let xrefUri = await xref2Uri( k, v );
  if( dbIds[k] != null ){
    dbIds[k] = dbIds[k].concat( v );
  } else {
    dbIds[k] = [v];
  }
});
```

Reply:
Good point - the Promise.all calls were resulting in a bunch of cache misses for the same db as the same xrefs dbs were being sent out at the same time. Made serial and inlined..